scico/linop/_linop.py

```diff
@@ -169,7 +169,7 @@ class Diagonal(LinearOperator):
 
     def __init__(
         self,
-        diagonal: Union[Array, BlockArray],
+        diagonal: Union[JaxArray, BlockArray],
```

Comment:
Perhaps this union could be defined in `scico.typing` as something like `array_like` or `vector`. Not sure how far we want to go with typing now.

Reply:
That occurred to me too. As a practical matter, though, this would probably introduce some messy circular imports since `blockarray` imports `typing`. Calling it `array_like` is a reasonable choice, but it's inconsistent with the naming scheme in `typing`. Perhaps `JaxOrBlockArray`?

Reply:
Agreed to return to this at some other time, with low priority.